pkg/validation/app.go

```diff
@@ -305,6 +313,10 @@ func (v *Validator) validateMetadataConstraints(ctx context.Context, cr v1alpha1
 }
 
 func (v *Validator) validateUserConfig(ctx context.Context, cr v1alpha1.App) error {
+	if key.IsManagedByFlux(cr, v.projectName) {
```

Comment:
I am not quite sure why we skip all of the validation? Maybe I am missing what changes with the flux setup here. I would have assumed that e.g. checking for existence of the CM is valid.

Reply:
@MarcelMue Yes, it's to make the flux setup more intuitive. We want to have the user values CM or secret be in the same directory as the app CR like [here](https://github.com/giantswarm/workload-clusters-fleet-demo/tree/ac17c5b9b6473b28d618839472226b3547311c56/management-clusters/gauss/workload-clusters/q4v5w/apps/hello-world-a).

But when kustomize renders the templates the app CR is applied first which fails due to the missing resources. @ljakimczuk couldn't find a way of specifying this dependency without having a separate directory for the user values and depending on it for the app CR.

With this approach we skip the CM and secret checks in app-admission-controller but keep them in the validation resource in app-operator. It sets the app CR status and delays the processing. Once flux has created the CM or secret the app CR will be processed and this should be quick so the user isn't affected.

```
status:
  release:
    reason: 'validation error: configmap `nginx-ingress-controller-app-user-values`
      in namespace `d5c8y` not found'
    status: resource-not-found
```

This looked fine in my testing yesterday but will do some more today. WDYT?